### PR TITLE
Adding postgresql-ha on obs cluster

### PR DESCRIPTION
**Pull request by RH-csaggin**
- created all the postgresql-ha config resources for the ai-telemetry project
- creation of external secrets
- creation of namespace

### CODE REVIEW

File: ai-telemetry/overlays/nerc-ocp-obs/externalsecrets/postgresql-ha-pgpool-externalsecret.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: postgresql-ha-pgpool-secret
```
Comment (computate) [inline ai-telemetry/overlays/nerc-ocp-obs/externalsecrets/postgresql-ha-pgpool-externalsecret.yaml:4]: Check that the name here matches the other part of this ExternalSecret.

File: ai-telemetry/base/externalsecrets/postgresql-ha-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: postresql-ha-secret
```
Comment (computate) [inline ai-telemetry/base/externalsecrets/postgresql-ha-secret.yaml:4]: Check that the name here matches the patch for this ExternalSecret.

**Review by computate — CHANGES_REQUESTED** [review]

File: ai-telemetry/base/externalsecrets/postgresql-ha-pgpool-secret.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: postgresql-ha-pgpool
```
Comment (computate) [inline ai-telemetry/base/externalsecrets/postgresql-ha-pgpool-secret.yaml]: Check that the name here matches the patch for this ExternalSecret.

**Review by computate — CHANGES_REQUESTED** [review]

**Comment by RH-csaggin** [comment]
@computate Well spotted I have one conflict name plus another typo. Now looks good.

File: ai-telemetry/base/postgresql-ha/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+            - name: POSTGRESQL_VOLUME_DIR
+              value: "/bitnami/postgresql"
+            - name: PGDATA
+              value: "/bitnami/postgresql/data"
+            - name: POSTGRES_USER
+              value: "postgres"
+            - name: POSTGRES_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: postgresql-ha-postgresql
```
Comment (computate) [inline ai-telemetry/base/postgresql-ha/statefulset.yaml:69]: This should be `postgresql-ha-secret`.

File: ai-telemetry/base/postgresql-ha/statefulset.yaml [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+            - name: REPMGR_RECONNECT_ATTEMPTS
+              value: "2"
+            - name: REPMGR_RECONNECT_INTERVAL
+              value: "3"
+            - name: REPMGR_USERNAME
+              value: "repmgr"
+            - name: REPMGR_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: postgresql-ha-postgresql
```
Comment (computate) [inline ai-telemetry/base/postgresql-ha/statefulset.yaml:131]: This should be `postgresql-ha-secret`.

File: ai-telemetry/base/postgresql-ha/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            - name: BITNAMI_DEBUG
+              value: "false"
+            - name: PGPOOL_BACKEND_NODES
+              value: 0:postgresql-ha-postgresql-0.postgresql-ha-postgresql-headless:5432,1:postgresql-ha-postgresql-1.postgresql-ha-postgresql-headless:5432,2:postgresql-ha-postgresql-2.postgresql-ha-postgresql-headless:5432,
+            - name: PGPOOL_SR_CHECK_USER
+              value: "repmgr"
+            - name: PGPOOL_SR_CHECK_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: postgresql-ha-postgresql
```
Comment (computate) [inline ai-telemetry/base/postgresql-ha/deployment.yaml:51]: This should be `postgresql-ha-secret`.

File: ai-telemetry/base/postgresql-ha/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+            - name: PGPOOL_SR_CHECK_DATABASE
+              value: "postgres"
+            - name: PGPOOL_ENABLE_LDAP
+              value: "no"
+            - name: PGPOOL_POSTGRES_USERNAME
+              value: "postgres"
+            - name: PGPOOL_POSTGRES_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: postgresql-ha-postgresql
```
Comment (computate) [inline ai-telemetry/base/postgresql-ha/deployment.yaml:62]: This should be `postgresql-ha-secret`.

File: ai-telemetry/base/externalsecrets/postgresql-ha-secret.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: postgresql-ha-secret
+  namespace: postgresql-ha
+spec:
+  refreshInterval: 15s
+  secretStoreRef:
+    name: nerc-cluster-secrets
+    kind: ClusterSecretStore
+  target:
+    name: postresql-ha-secret
```
Comment (computate) [inline ai-telemetry/base/externalsecrets/postgresql-ha-secret.yaml:12]: This should be `postgresql-ha-secret`.

**Review by computate — CHANGES_REQUESTED** [review]

**Comment by computate** [comment]
Replacing this PR [with another PR](https://github.com/OCP-on-NERC/nerc-ocp-config/pull/597) containing the latest from the main branch, and some fixes to the secret names.

**Comment by computate** [comment]
Closing this, we will use Crunchy Postgres instead.

**Pull request closed without merging**